pkg/apis/core/validation/events.go

```diff
@@ -33,8 +37,83 @@ const (
 	NoteLengthLimit              = 1024
 )
 
-// ValidateEvent makes sure that the event makes sense.
-func ValidateEvent(event *core.Event) field.ErrorList {
+func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure events always pass legacy validation.
+	legacyErrs := legacyValidateEvent(event)
+	if len(legacyErrs) > 0 {
+		return legacyErrs
+	}
+
+	if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
+		// No further validation for backwards compatibility.
+		return nil
+	}
+
+	// Strict validation applies to creation via events.k8s.io/v1 API and newer.
+	allErrs := ValidateObjectMeta(&event.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateV1EventSpec(event)...)
+	return allErrs
+}
+
+func ValidateEventUpdate(newEvent, oldEvent *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure the new event always passes legacy validation.
+	legacyErrs := legacyValidateEvent(newEvent)
+	if len(legacyErrs) > 0 {
+		return legacyErrs
```

Comment:
Same comment as for ValidateEventCreate.

Reply:
Done.